software/utils/debug/debug_cli.py

```diff
@@ -0,0 +1,645 @@
+            self.interface.variables_update_info()
+            self.update_prompt()
+
+        else:
+            self.interface.connect(params[0])
+            self.interface.resynchronize()
+            self.interface.variables_update_info()
+            self.update_prompt()
+
+    def help_run(self):
+        print(
+            f"""\
+Run an external Python script which can send commands, set variables, etc.
+
+If no explicit path is given then the current directory and a sub-directory
+named {self.scripts_directory} will be searched for the python script.
+        """
+        )
+
+    def do_run(self, line):
+        params = shlex.split(line)
+
+        if len(params) < 1:
+            red("Must specify script name for `run` command")
+            return
+
+        if not os.path.exists(self.scripts_directory + "/" + params[0] + ".py"):
+            print("Unknown script " + params[0])
+            return
+
+        import_command = f"import {self.scripts_directory}.{params[0]}"
+        exec(import_command)
+
+        passed_params = ""
+        if len(params) > 1:
+            passed_params = ', "' + " ".join(params[1:]) + '"'
+        run_command = f"{self.scripts_directory}.{params[0]}.{params[0]}(self.interface{passed_params})"
+        exec(run_command)
+
+    def complete_run(self, text, line, begidx, endidx):
+        return [
+            x[len(self.scripts_directory) + 1 : len(x) - 3]
+            for x in glob.glob(self.scripts_directory + "/" + text + "*.py")
+        ]
+
+    def do_test(self, line):
+        """This command is for working with structured test scenarios and saved test data.
+
+A test scenario defines independent variables in question, whether they be ventilator settings
+or manual settings applicable to external test equipment. It may also specify test criteria
+which can be confirmed by human examination or other scripts. A scenario also defines which
+variables to capture, how often to sample them and length of test run.
+
+Running a test scenario will capture traced variables as well as a snapshot of all ventilator
+variables and other available metadata concerning the testing environment.
+
+The unique test identifier (as well as the file name) is defined as
+    <UTC date-time>_<tester>_<scenario_name>
+
+test load <file>
+  loads test scenarios from specified .csv or .json <file>
+
+test autoload
+  loads all test scenarios found in `test_scenarios` subdirectory
+
+test clear
+  clears all test scenarios loaded in this session
+
+test list [--verbose/-v]
+  lists all test scenarios loaded in this session
+
+test show <scenario>
+  prints out the full definition of named <scenario>
+
+test apply <scenario>
+  applies all ventilator settings for named <scenario>
+
+test run <scenario> [--verbose/-v] [--plot/-p]
+  runs named <scenario> and saves data to .json
+    --verbose/-v  will also print out full trace data in columns
+    --plot/-p will also plot traces and save plots as .png
```

Comment:
No, you do not need to apply it. Running it applies whatever settings can be applied automatically. This may just be used to temporarily "adopt" the prescribed settings for manual experimentation. Not sure how to compactly convey this in the help...